js/util/util.js

```diff
@@ -400,3 +400,15 @@ exports.getCoordinatesCenter = function(coords) {
     return new Coordinate((minX + maxX) / 2, (minY + maxY) / 2, 0)
         .zoomTo(Math.floor(-Math.log(dMax) / Math.LN2));
 };
+
+/**
+ * Throw an error if a condition is not met
+ * @param {boolean} condition
+ * @returns {?string} message
+ * @private
+ */
+exports.assert = function(condition, message) {
+    if (!condition) {
+        throw new Error(message || 'Assertion failed');
+    }
+};
```

Comment:
Looks like `console.assert` is not and will not be widely supported. `util.assert` is just a shorthand around `if (blah) throw new Error(...)`. Validating preconditions is especially helpful because WebGL is bad at reporting errors. I can remove the `util.assert` statements before merging this code if you prefer.